### ISSUE TITLE
MSSQL - strange behavior with limit and already specified order

### ISSUE THREAD

**Issue by arnaud-moncel** [sequelize/sequelize#14185]
## Issue Creation Checklist

- [X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description

Hi guys 👋, I just want to confirm with you if I found a bug or if I just do something wrong.

Let's say I have 2 models defined like this:
```js
const City = sequelize.define(
  'city',
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
      allowNull: false,
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
  },
  {
    tableName: 'city',
    timestamps: false,
  },
);

const Address = sequelize.define(
  `address`,
  {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
      allowNull: false,
    },
    postalCode: {
      type: DataTypes.STRING,
    },
  },
  {
    tableName: 'address',
    timestamps: false,
  },
);

City.hasMany(Address);
Address.belongsTo(City);
```

I have this findAll query 👇, the goal is pretty simple, I just want to make a leaderboard by the most used `city` in `addresses`. 
```js
const query = {
  attributes: [
    [col('city.name'), 'city__grouped__'],
    [fn('count', '*'), '__aggregate__'],
  ],
  group: ['city.name'],
  include: [
    {
      association: 'addresses',
      attributes: [],
      include: [],
    },
  ],
  limit: 5,
  order: [[fn('count', '*'), 'DESC']],
  subQuery: false,
  raw: true,
};
```

This is where you can help me determine if it's a bug or if I did something wrong.
The call to the findAll function result to this query 👇
```sql
SELECT
  [city].[name] AS [city__grouped__],
  COUNT(*) AS [__aggregate__]
FROM
  [city] AS [city]
  LEFT OUTER JOIN [address] AS [addresses] ON [city].[id] = [ddresses].[cityId]
GROUP BY
  [city].[name]
ORDER BY
  COUNT(*) DESC,
  [city].[id] OFFSET 0 ROWS FETCH NEXT 5 ROWS ONLY;
```
In fact this query result to a SQL error that say:
`Column "city.city_id" is invalid in the ORDER BY clause because it is not contained in either an aggregate function or the GROUP BY clause.`

I deep dive in the code and I found something strange here https://github.com/sequelize/sequelize/blob/abdb8967ef1fa458e1e36cd641ea140874409e7d/src/dialects/mssql/query-generator.js#L993-L1006

Sorry about the long message, but my question is: **Why [city].[id] is added if I specify an order in my query?** 

let me know if it not clearer enough.
Thank's for your time 🙏 have a great day.

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [X] Yes. This issue only applies to the following dialect(s): MSSQL
- [ ] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [X] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
That's weird.

The only line in mssql's query generator that mutates the order option is this one https://github.com/sequelize/sequelize/blob/main/src/dialects/mssql/query-generator.js#L881 
I'm not fully convinced it's the culprit. It would help to determine if this issue happens with other dialects as well

**Comment by ephys** [comment]
Aah, sorry I went too fast again and didn't read carefully enough

I remember seeing the weird behavior in that function too, I added the comment just before it https://github.com/sequelize/sequelize/blob/abdb8967ef1fa458e1e36cd641ea140874409e7d/src/dialects/mssql/query-generator.js#L991

Maybe mssql had (has?) a requirement to specify `ORDER BY` when using limit/offset. Or it was a decision from the author of the previous maintainers.

Anyway, it seems the reason the clause is added is because this piece of code did not expect `subQuery` to be false. 
If you're willing to open a PR with a test & a fix, we'll try to review it in a timely matter :)

**Comment by arnaud-moncel** [comment]
You are right, and I just tested it mssql need an `order by` to be able to use `offset` ans `fetch` features.
I'm gonna open a PR to fix this thank's again.

**Comment by aniruth37** [comment]
I saw that this has been stale for a while. Raised a PR of v6 - https://github.com/sequelize/sequelize/pull/16776.

**Comment by rishabh-arya95** [comment]
@ephys This issue still happening, can anyone review the PR - #16776
